src/main/java/com/persado/oss/quality/stevia/selenium/core/controllers/webdriverapi/BySizzle.java

```diff
@@ -38,115 +38,68 @@
 
 
 import com.persado.oss.quality.stevia.selenium.core.SteviaContext;
-import com.persado.oss.quality.stevia.selenium.core.WebController;
-import com.persado.oss.quality.stevia.selenium.core.controllers.AppiumWebController;
 import com.persado.oss.quality.stevia.selenium.core.controllers.WebDriverWebController;
 import org.openqa.selenium.*;
-import org.openqa.selenium.internal.FindsByCssSelector;
-import org.openqa.selenium.internal.FindsByXPath;
 import org.openqa.selenium.remote.RemoteWebElement;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.io.Serializable;
 import java.lang.reflect.Method;
 import java.util.Collections;
 import java.util.List;
-import java.util.regex.Pattern;
 
-public abstract class ByExtended extends By {
+public abstract class BySizzle extends By {
 
-    private static final Logger LOG = LoggerFactory.getLogger(ByExtended.class);
+    private static final Logger LOG = LoggerFactory.getLogger(BySizzle.class);
 
     /**
      * Finds elements via the driver's underlying W3 Selector engine. If the
      * browser does not implement the Selector API, a best effort is made to
      * emulate the API. In this case, we strive for at least CSS2 support, but
      * offer no guarantees.
+     *
+     * @param sizzleCssSelector selector
+     * @return css selector
      */
-    public static By cssSelector(final String selector) {
-        if (selector == null)
+    public static By css(final String sizzleCssSelector) {
+        if (sizzleCssSelector == null)
             throw new IllegalArgumentException(
                     "Cannot find elements when the selector is null");
 
-        return new ByCssSelectorExtended(selector);
+        return new BySizzleCssSelector(sizzleCssSelector);
 
     }
 
-    /**
-     * @param xpathExpression The xpath to use
-     * @return a By which locates elements via XPath
-     */
-    public static By xpath(final String xpathExpression) {
-        if (xpathExpression == null)
-            throw new IllegalArgumentException(
-                    "Cannot find elements when the XPath expression is null.");
-
-        return new ByXPathExtended(xpathExpression);
-    }
```

Comment:
Why did you remove the xpath related code? Are the xpath selectors handled in another class?

Reply:
This class is only used for sizzle related locators, for xpath native By class is used
https://github.com/Workable/stevia/blob/634b3edd2e823c09e96cf49d4b8358003c7b0a6b/src/main/java/com/persado/oss/quality/stevia/selenium/core/controllers/WebDriverWebController.java#L239